tests/Crypto/tst_cryptorequests/tst_cryptorequests.cpp

```diff
@@ -299,6 +304,13 @@ void tst_cryptorequests::generateKeyEncryptDecrypt()
     // test encrypting some plaintext with the generated key
     QByteArray plaintext = "Test plaintext data";
     QByteArray initVector = "0123456789abcdef";
+    QByteArray authData("fedcba9876543210");
+    QByteArray tag;
+
+    if (blockMode == CryptoManager::BlockModeGcm) {
+        initVector.resize(12);  // GCM default IV size is 12
```

Comment:
Why is this required for these tests, but not the cipherEncryptDecrypt() one?

Reply:
I think that is a bug in the cipherEncryptDecrypt() case. It should check the IV size as well. Will look into that.

Reply:
Ah actually, it's because OpenSslCryptoPlugin::initialiseCipherSession() automatically generates a new IV if the specified IV has an incorrect size. I think we discussed that Encrypt/DecryptRequest would just use the given IV instead of trying to correct it, but we should probably have a consistent approach across both methods?

Reply:
When it generates a new IV, how does the user get notified about the new IV?

Reply:
For CipherRequest, the IV can be checked via CipherRequest::generatedInitialisationVector.

Reply:
I agree that we should be consistent across both methods and offer generatedInitialisationVector() for both.  However, perhaps the semantics of that should be: if any IV is provided, but is wrong, return an error; only generate an IV if an empty byte array is provided as the input IV.  Does this sound sensible or are there good reasons not to do this?

Reply:
What if we have a generateInitializationVector() or such, so that the user can explicitly assign one? That way there is no uncertainty about whether one has been automatically generated.

(Btw I noticed we sometimes use "initialise" and other times "initialize", so that should be standardized soonish, presumably to the US "initialize" spelling.)

Reply:
Yes, we should standardise to American english.

I don't think we should have a separate generateInitializationVector operation/request - it seems far more intuitive/simple to me to have it generated as part of the encrypt request, honestly (mainly because it's not a separate thing - the IV is part of the encrypt operation.  We provide a generateRandomData request for the case where they want to explicitly generate an IV separately and set it).  What are your reasons for preferring a separate request for that?